modules/EmergencyMedicine/src/EmergencyMedicine/Controller/BaseController.php

```diff
@@ -0,0 +1,121 @@
+    const DOC_TYPE = "file_url";
+    const PDF_MINE_TYPE = "application/pdf";
+
+    private $container;
+
+    public function __construct(ContainerInterface $container)
+    {
+        parent::__construct($container);
+        $this->container = $container;
+    }
+
+    /**
+     * get facility data
+     */
+    public function getFacilityInfo($id = null)
+    {
+        if (!is_null($id)) {
+            $info = $this->container->get('GenericTools\Model\FacilityTable')->getFacility(intval($id));
+            $data = array();
+            $data['clinic'] = $info->name;
+            $data['address'] = $info->street . " " . $info->city;
+            $data['phone'] = $info->phone;
+            $data['email'] = $info->email;
+            return $data;
+
+        } else {
+            return array();
+        }
+    }
+
+    public function saveDocToStorage($data, $fileName, $date)
+    {
+        $dataToSave = array();
+        $dataToSave['storage']['data'] = $data;
+        $dataToSave['documents']['date'] = $date;
+        $dataToSave['documents']['url'] = $fileName;
+        $rez = $this->uploadToStorage($dataToSave);
+        return $rez;
+    }
+
+    public function saveDocInfoToDb($storageSave, $configData,$pdfEncoded)
+    {
+
+        if ($storageSave['id']) {
+
+            $configData = array_merge($configData, $storageSave);
+
+            $dbStructuredData = $this->buildArrToDb($configData);
+
+            if (empty($dbStructuredData)) {
+                ErrorCodes::http_response_code('500', 'failed to build data to db');
+                return array();
+            } else {
+
+                $save = $this->saveDocToDb($dbStructuredData);
+
+                if($save){
+                    return array(
+                        "id" => $save,
+                        "base64_data" => $pdfEncoded
+                    );
+                }else{
+                    ErrorCodes::http_response_code('500', 'failed to build data to db');
+                    return array();
+                }
+            }
+        } else {
+            ErrorCodes::http_response_code('500', 'failed to save document');
+            return array();
+        }
+    }
+
+    public function createConfigData($postData,$mimetype, $category)
+    {
+        if (empty($postData['facility']) || empty($postData['encounter'])) {
+            return array();
+        }
+
+        $configData = array(
+            'mimetype' => $mimetype,
+            'category' => $category,
+            'encounter' => $postData['encounter']
+        );
+
+        if (empty($postData['owner'])) {
+            $configData['owner'] = $postData['owner'];
+        }
+
+        if (empty($postData['patient'])) {
+            $configData['patient'] = $postData['patient'];
+        }
+        return $configData;
+    }
+
+    public function createBase64Pdf($fileName,$headerPath, $footerPath,$headerData,$bodyData)
```

Comment:
Why do not move this function to GenericTools\Traits\saveDocToServer

Reply:
it cannot be traint since it uses $this, and does a spesific set of actions, I don't see the use of it in a stand alone trait